### PR TITLE
feat(ui): Connect Activity list API

### PR DESCRIPTION
**Pull request by billyvg**
Some fixes to get Activity list component connected to API

Depends on https://github.com/getsentry/sentry/pull/13291, https://github.com/getsentry/sentry/pull/13294

### CODE REVIEW

File: src/sentry/static/sentry/app/components/activity/note/index.jsx [hunk 1]
```diff
@@ -50,9 +60,9 @@ class Note extends React.Component {
   };
 
   handleDelete = () => {
-    const {item, onDelete} = this.props;
+    const {onDelete} = this.props;
 
-    onDelete(item);
+    onDelete(this.props);
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/activity/note/index.jsx]: Is this intentional? We probably don't want to call this with *all* of the props

Comment (billyvg) [reply src/sentry/static/sentry/app/components/activity/note/index.jsx]: Yeah, this component doesn’t need to make a decision on what the call handler needs.

**Review by lynnagara — APPROVED** [review]
Seems like most of that code didn't end up being part of this PR anyway?

File: src/sentry/static/sentry/app/actionCreators/incident.jsx [hunk 2]
```diff
@@ -46,7 +46,9 @@ export async function createIncident(
 /**
  * Fetches a list of activities for an incident
  */
-export async function fetchIncidentActivities(api, orgId, incidentId) {}
+export async function fetchIncidentActivities(api, orgId, incidentId) {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/actionCreators/incident.jsx]: Should all of our API actionCreators be async functions like this?

**Comment by billyvg** [comment]
Ah yeah, sorry forgot what code was in which PR.

File: src/sentry/static/sentry/app/actionCreators/incident.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/actionCreators/incident.jsx]: In an ideal world components wouldn't be able to listen to them (instead they'd have to only listen to a store) - I made it async here to be consistent with the rest (which `await` to display status messages).

**Pull request merged**